fw/ss_skb.c

```diff
@@ -1008,6 +1008,108 @@ ss_skb_chop_head_tail(struct sk_buff *skb_head, struct sk_buff *skb,
 	return 0;
 }
 
+/**
+ * Chop @head and @trail bytes at head and end of the message (skb circular
+ * list), iterating over the list if nessessary.
+ * @return 0 on success and -(error code) on error.
+ *
+ * Message length should be greater than the sum of @head and @trail,
+ * otherwise an error is retuned.
+ */
+int
+ss_skb_list_chop_head_tail(struct sk_buff **skb_list_head,
+			   size_t head, size_t trail)
+{
+	struct sk_buff *skb, *skb_hd;
+	size_t sum;
+	int ret;
+
+	skb_hd = *skb_list_head;
+	if (likely(skb_hd->next == skb_hd))
+		goto single_buff;
+	goto multi_buffs;
+
+	/* Everywhere in the function we perform 'redundant'
+	 * checks for @head and @trail values to avoid unneccessary
+	 * calls to underlying ss_skb_chop_head_tail() for
+	 * optimization purpose
+	 */
+
+single_buff:
+	/* There is the only 1 buffer in the list
+	 * and skb_hd points to it
+	 */
+	sum = head + trail;
+	if (WARN_ON_ONCE(skb_hd->len <= sum))
+		return -EBADMSG;
+	if (unlikely(sum == 0))
+		/* Nothing to chop */
+		/* This check is mostly for jumps from branches below */
+		return 0;
+	return ss_skb_chop_head_tail(NULL, skb_hd, head, trail);
+
+multi_buffs:
+	/* skb_list contains more than 1 skb &&
+	 * skb_hd points to head element of the list
+	 */
```

Comment:
I think there is no sense to copy and paste the comment 3 times: just comment it in one place and reader will get the idea for the other places

Reply:
@krizhanovsky 
These comments are not exactly the same.
The main idea is that they are formal preconditions for the code below.
They make the future maintenance of the code more easy, including other programmers.
So let they remain as is.